packages/node_modules/@ciscospark/plugin-lyra/test/integration/spec/device.js

```diff
@@ -0,0 +1,170 @@
+            muted: false
+          },
+          deviceUrl: lyraMachine.spark.device.url
+        };
+
+        return lyraMachine.spark.lyra.device.putAudioState(lyraMachine.space, audioState);
+      });
+
+      it(`returns audio state`, () => lyraMachine.spark.lyra.device.getAudioState(lyraMachine.space)
+          .then((res) => {
+            assert.equal(res.microphones.muted, audioState.microphones.muted);
+            assert.equal(res.volume.level, audioState.volume.level);
+          }));
+    });
+
+    describe(`when a call is in progress`, () => {
+      before(`ensure participant joined space`, () => spock.spark.lyra.space.join(lyraMachine.space)
+        .then(() => lyraMachine.spark.lyra.space.verifyOccupant(lyraMachine.space, spock.id))
+        .then(() => spock.spark.lyra.space.bindConversation(lyraMachine.space, conversation)));
+
+      before(`make a call`, () => {
+        const locus = {
+          url: conversation.locusUrl,
+          correlationId: uuid.v4()
+        };
+
+        return spock.spark.request({
+          method: `POST`,
+          uri: `${locus.url}/participant`,
+          body: {
+            correlationId: locus.correlationId,
+            deviceUrl: spock.spark.device.url,
+            localMedias: []
+          }
+        });
+      });
+
+      after(`remove binding`, () => spock.spark.lyra.space.unbindConversation(lyraMachine.space, conversation));
+
+      it(`allows muting`, (done) => {
+        lyraMachine.spark.mercury.once(`event:lyra.space_audio_microphones_mute_action`, (event) => {
+          assert.equal(event.data.action, `mute`);
+          done();
+        });
+
+        return spock.spark.lyra.device.mute(lyraMachine.space);
+      });
+
+      it(`allows unmuting`, (done) => {
+        lyraMachine.spark.mercury.once(`event:lyra.space_audio_microphones_mute_action`, (event) => {
+          assert.equal(event.data.action, `unMute`);
+          done();
+        });
+
+        return spock.spark.lyra.device.unmute(lyraMachine.space);
+      });
+
+      it(`allows increasing volume`, (done) => {
+        lyraMachine.spark.mercury.once(`event:lyra.space_audio_volume_change_action`, (event) => {
+          assert.equal(event.data.action, `increase`);
+          done();
+        });
+
+        return spock.spark.lyra.device.increaseVolume(lyraMachine.space);
+      });
+
+      it(`allows decreasing volume`, (done) => {
+        lyraMachine.spark.mercury.once(`event:lyra.space_audio_volume_change_action`, (event) => {
+          assert.equal(event.data.action, `decrease`);
+          done();
+        });
+
+        return spock.spark.lyra.device.decreaseVolume(lyraMachine.space);
+      });
+
+      it(`allows setting volumne`, (done) => {
+        lyraMachine.spark.mercury.once(`event:lyra.space_audio_volume_set_action`, (event) => {
+          assert.equal(event.data.level, 2);
+          done();
+        });
+
+        return spock.spark.lyra.device.setVolume(lyraMachine.space, 2);
```

Comment:
There's a helper in the phone plugin for waiting for events in a promisy fashion